### PR TITLE
Unit tests

### LINKED ISSUE #2: Testing
Testing grammar files is a little tricky, the best approach I can think of is to find a utility which can generate formatted HTML using the reference files and this grammar.

We could then compare reference (expected) output against the generated. This would help protect against accidental breakages.

Better still if we can break down the reference files into smaller chunks to give us a pass/fail on the different syntax types.

There may be some visual diff algorithms for HTML files which might make evaluating changes easier:

https://github.com/search?q=html+diff+visual&type=Repositories

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #2

### CODE REVIEW

**Comment by MetRonnie** [comment]
I didn't expect GitHub to run the workflow from this PR branch rather than the master workflow... in that case I can push fixes to this PR instead of a new one

**Comment by oliver-sanders** [comment]
Really nice testing framework, only a couple of tests failing, good to go once they are sorted. If you could squash down your commits at some point that would help.

> I didn't expect GitHub Actions to run the workflow from this PR branch rather than the master workflow

Actions (and Travis) will always run off of the PR branch. Note that actions is testing the branch not the changes so if you fix something on master you will need to rebase your branch to get the action to pass.

File: tests/suite-test-1.2.rc [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# SYNTAX TEST "source.cylc" "Comments"
+# See https://github.com/cylc/cylc-flow/issues/2752
+
```
Comment (wxtim) [inline tests/suite-test-1.2.rc:3]: Looks like you've separated your tests into logical blocks - and it's a great framework to read.

I think that the name of the test files should give more of a clue about what they are testing.
Failing (or in addition to) that, I'd like a little more comment in the top of the file.

```suggestion
# Testing that comments are correctly identified.
```

Comment (MetRonnie) [reply tests/suite-test-1.2.rc:3]: Yes I did wonder the same

**Pull request merged**